### PR TITLE
Make abs_phase default to True if TZRMJD is set

### PR DESCRIPTION
**Pull request by paulray**
Fixes issue #1240 for @matteobachetti

### CODE REVIEW

**Comment by aarchiba** [comment]
Looks good to me. Blacken and confirm it passes CI and I think it's ready to merge.

**Comment by matteobachetti** [comment]
Fantastic! Thanks!

**Comment by paulray** [comment]
Sadly, this seem to break many many tests.

**Comment by paulray** [comment]
Unfortunately not so simple. It breaks all(?) the derivative tests, for reasons I don't yet understand. So, there isn't an explicit call to `model.phase()` where I can add `abs_phase=False`.

File: tests/test_model_derivatives.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def f(value):
         with quiet():
             warnings.simplefilter("ignore")
             try:
-                dphase = m.phase(toas) - phase
+                dphase = m.phase(toas, abs_phase=False) - phase
```
[the remark this answers is not in the document]
Comment (paulray) [reply tests/test_model_derivatives.py:157]: I don't know. With my changes, the tests all pass, but that leaves me concerned that there is an issue computing derivatives for models with AbsPhase components.

[the remark this answers is not in the document]
Comment (aarchiba) [reply tests/test_model_derivatives.py:157]: I guess a mean-subtracted derivative is different from a non-mean-subtracted derivative? I wonder where that is implemented, though? Like if you're taking the derivative associated with a JUMP, is it all zeros and ones?

[the remark this answers is not in the document]
Comment (aarchiba) [reply tests/test_model_derivatives.py:157]: This question about derivatives isn't the result of your PR - anyone could have asked for absolute phase and run into it earlier. If you want to kick the can down the road a bit, maybe just file an issue?

**Comment by aarchiba** [comment]
Could you please create an issue describing the problem you see with the derivatives? Then we can merge this.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -6,6 +6,6 @@ repos:
     -   id: check-merge-conflict
     -   id: check-symlinks
 -   repo: https://github.com/psf/black
-    rev: 19.3b0
+    rev: 22.3.0
```
Comment (paulray) [inline .pre-commit-config.yaml:9]: @aarchiba I had to make this change to get my pre-commit to work. I think this is the correct thing now that we've moved to the new black.

**Comment by aarchiba** [comment]
Could you please write a CHANGELOG entry? Probably a good idea to merge/rebase first to avoid annoying conflicts in CHANGELOG itself.

**Comment by paulray** [comment]
There is a CHANGELOG entry and there seem to be no conflicts.

**Pull request merged**